### PR TITLE
Add uncrustify to make system

### PR DESCRIPTION
**Pull request by mlyle**
Should supplant astyle, but astyle is not removed in this PR.

### CODE REVIEW

File: make/tools.mk [hunk 1]
```diff
@@ -439,6 +439,45 @@ astyle_clean:
 	$(V0) @echo " CLEAN        $(ASTYLE_BUILD_DIR)"
 	$(V1) [ ! -d "$(ASTYLE_BUILD_DIR)" ] || $(RM) -r "$(ASTYLE_BUILD_DIR)"
 
+# Set up uncrustify tools
+UNCRUSTIFY_DIR := $(TOOLS_DIR)/uncrustify-0.61
+UNCRUSTIFY_BUILD_DIR := $(DL_DIR)/uncrustify
+libusb_win_install: LIBUSB_WIN_FILE := libusb-win32-bin-1.2.6.0.zip
```
Comment (mluessi) [inline make/tools.mk]: Should this be here (doesn't seem to be related to uncrustify)

File: make/uncrustify.cfg [hunk 2]
```diff
@@ -0,0 +1,148 @@
+# first attempt at an uncrustify config file for dRonin
```
Comment (mluessi) [inline make/uncrustify.cfg]: ;)

File: make/tools.mk [hunk 3]
```diff
@@ -439,6 +439,45 @@ astyle_clean:
...
+
+.PHONY: uncrustify_install
+uncrustify_install: | $(DL_DIR) $(TOOLS_DIR)
+uncrustify_install: UNCRUSTIFY_URL := http://downloads.sourceforge.net/project/uncrustify/uncrustify/uncrustify-0.61/uncrustify-0.61.tar.gz
```
Comment (PTDreamer) [inline make/tools.mk]: Wouldn't it be better to follow the project's gitHub master branch instead of having to update this on every new release?

File: make/tools.mk [hunk 4]
```diff
@@ -439,6 +439,45 @@ astyle_clean:
 	$(V0) @echo " CLEAN        $(ASTYLE_BUILD_DIR)"
 	$(V1) [ ! -d "$(ASTYLE_BUILD_DIR)" ] || $(RM) -r "$(ASTYLE_BUILD_DIR)"
 
+# Set up uncrustify tools
+UNCRUSTIFY_DIR := $(TOOLS_DIR)/uncrustify-0.61
+UNCRUSTIFY_BUILD_DIR := $(DL_DIR)/uncrustify
+libusb_win_install: LIBUSB_WIN_FILE := libusb-win32-bin-1.2.6.0.zip
+
+.PHONY: uncrustify_install
```
Comment (PTDreamer) [inline make/tools.mk]: new target was not added to top level makefile help

File: make/tools.mk (hunk 3 above)
Comment (mlyle) [inline make/tools.mk]: I don't think so-- I think things should be reproducible and we should only move between versions of tools if we really mean to.  If they change something that alters interpretation of the config file, etc, we could break.

File: make/tools.mk (hunk 4 above)
Comment (mlyle) [inline make/tools.mk]: Added to help, and marked astyle as 'deprecated' in the help.

File: make/uncrustify.cfg [hunk 5]
```diff
@@ -0,0 +1,147 @@
+# run with FLIGHT=some_file.c make uncrustify_flight
+
```
Comment (kubark42) [inline make/uncrustify.cfg]: @mlyle do you know of a handy online calculator for visualizing the effects of this uncrustify configuration?

Comment (mlyle) [inline make/uncrustify.cfg]: There's a program called 'universalindentgui' that you can load this into and live indent a source file.
